debug_module.tex

```diff
@@ -245,20 +245,24 @@ \subsection{Abstract Command Listing}
 \section{Program Buffer} \label{programbuffer}
 
 To support executing arbitrary instructions on a halted hart,
-a Debug Module can include a 
-Program Buffer that a debugger can write small programs to. Systems
+a Debug Module can include a Program Buffer that a debugger
+can write small programs to. Systems
 that support all necessary functionality using abstract commands
 only may choose to omit the Program Buffer.
 
 A debugger can write a small program to the Program Buffer, and then
 execute it exactly once with the Access Register Abstract Command,
-setting the \Fpostexec bit in \Rcommand.
-If \Fprogsize is 1, the Program Buffer may only hold a single instruction.
-This can be a 32-bit
+setting the \Fpostexec bit in \Rcommand. The program in the 
+program buffer must end with {\tt ebreak} or {\tt c.ebreak}.
+If \Fprogbufsize is 1, the final {\tt ebreak} is implied and 
+the debugger need not write it. The contents of \Rprogbufzero
+can be a 32-bit
 instruction, or a compressed instruction in the lower 16 bits accompanied by a
-compressed {\tt nop} in the upper 16 bits.  If \Fprogsize is greater than 1,
-the debugger can write whatever program it likes, but the program must end with
-{\tt ebreak} or {\tt ebreak.c}.
+{\tt c.nop} in the upper 16 bits. 
+If \Fprogbufsize is greater than 1,
```

Comment:
Isn't this just saying the same as the above "The program in the program buffer must end with {\tt ebreak} or {\tt c.ebreak}."